### ISSUE TITLE
curl/libcurl getting pinned too specifically

### ISSUE THREAD

**Issue by xylar** [conda-forge/conda-forge-pinning-feedstock#537]
I believe that #157 incorrectly specified both major an minor versions for curl and libcurl even though they have:
```
 pin_run_as_build:
  curl:
    max_pin: x
  libcurl:
    max_pin: x
```
This seems to be causing problems in a rebuild of `nco`:  https://github.com/conda-forge/nco-feedstock/pull/108

**Comment by xylar** [comment]
If I manually change the build back to `curl 7` in a local build, things work as expected.

**Comment by isuruf** [comment]
I've always wanted to do this, but there was push back. Maybe third time is the charm?

**Comment by xylar** [comment]
I don't know.  I think we either need `pin_run_as_build` to be `x.x` or to just specify `7` as the pin because I'm seeing weird stuff with the current pins.

**Comment by isuruf** [comment]
> I think we either need pin_run_as_build to be x.x

Definitely not.

> just specify 7 as the pin because I'm seeing weird stuff with the current pins.

Or keep increasing the minor version.

**Comment by xylar** [comment]
For example, `libnetcdf` has the pin `curl 7.64` but then:
```
$ conda search --info libnetcdf=4.7.4
...
dependencies: 
...
  - curl >=7.64.1,<8.0a0
```
Seems weird and causes issues.

**Comment by isuruf** [comment]
`curl 7.64` and `curl >=7.64.1,<8.0a0` shouldn't conflict. `curl 7.64` is equivalent to `curl 7.64.*`

**Comment by xylar** [comment]
I think I see the problem.  @ocefpaf rebuilt `curl 7.64.0` with the latest `kbr5` pin but we need to rebuild `curl 7.64.1` with this pin instead.  Sorry for the noise here.  Problem needs to be solved on the `curl` feedstock.

**Comment by isuruf** [comment]
Looks like we need to keep building old versions of `libcurl` with new pinnings if we kept `major.minor`. So I'm fine with changing the version to just `major`

**Comment by xylar** [comment]
This has been addressed by #542
